src/main/java/net/minecraftforge/client/model/IModelBuilder.java

```diff
@@ -6,50 +6,103 @@
+    static IModelBuilder<?> of(boolean hasAmbientOcclusion, boolean usesBlockLight, boolean isGui3d,
+                               ItemTransforms transforms, ItemOverrides overrides, TextureAtlasSprite particle,
+                               RenderTypeGroup renderTypes)
+    {
+        return new Simple(hasAmbientOcclusion, usesBlockLight, isGui3d, transforms, overrides, particle, renderTypes);
+    }
+
+    static IModelBuilder<?> collecting(List<BakedQuad> quads)
     {
-        return new Simple(new SimpleBakedModel.Builder(owner, overrides).particle(particle));
+        return new Collecting(quads);
     }
 
-    T addFaceQuad(Direction facing, BakedQuad quad);
-    T addGeneralQuad(BakedQuad quad);
+    T addCulledFace(Direction facing, BakedQuad quad);
+
+    T addUnculledFace(BakedQuad quad);
 
     BakedModel build();
 
-    class Simple implements IModelBuilder<Simple> {
-        final SimpleBakedModel.Builder builder;
+    class Simple implements IModelBuilder<Simple>
+    {
+        private final SimpleBakedModel.Builder builder;
+        private final RenderTypeGroup renderTypes;
 
-        Simple(SimpleBakedModel.Builder builder)
+        private Simple(boolean hasAmbientOcclusion, boolean usesBlockLight, boolean isGui3d,
+                       ItemTransforms transforms, ItemOverrides overrides, TextureAtlasSprite particle,
+                       RenderTypeGroup renderTypes)
         {
-            this.builder = builder;
+            this.builder = new SimpleBakedModel.Builder(hasAmbientOcclusion, usesBlockLight, isGui3d, transforms, overrides).particle(particle);
+            this.renderTypes = renderTypes;
         }
 
         @Override
-        public Simple addFaceQuad(Direction facing, BakedQuad quad)
+        public Simple addCulledFace(Direction facing, BakedQuad quad)
         {
             builder.addCulledFace(facing, quad);
             return this;
         }
 
         @Override
-        public Simple addGeneralQuad(BakedQuad quad)
+        public Simple addUnculledFace(BakedQuad quad)
         {
             builder.addUnculledFace(quad);
             return this;
         }
 
+        @Deprecated
+        @Override
+        public BakedModel build()
+        {
+            return builder.build(renderTypes);
+        }
+    }
+
+    class Collecting implements IModelBuilder<Collecting>
+    {
+        private final List<BakedQuad> quads;
+
+        public Collecting(List<BakedQuad> quads)
+        {
+            this.quads = quads;
+        }
+
+        @Override
+        public Collecting addCulledFace(Direction facing, BakedQuad quad)
+        {
+            quads.add(quad);
+            return this;
+        }
+
+        @Override
+        public Collecting addUnculledFace(BakedQuad quad)
+        {
+            quads.add(quad);
+            return this;
+        }
+
         @Override
         public BakedModel build()
         {
-            return builder.build();
+            return EmptyModel.BAKED;
```

Comment:
It collects quads to a user-provided list, but it lacks any of the context to actually build it into a model that the other built-in model has. I might want to document the factory method to say that it will always return an empty model, though.

Reply:
A doc comment would definitely help, this is a bit unintuitive as it stands.

Reply:
Added in [f48934a](https://github.com/MinecraftForge/MinecraftForge/pull/8786/commits/f48934a8488de1c8fa613f4b21d72de611286c6e)